openedx/frontend-app-course-authoring
# feat: Add lib v2/legacy tabs in studio home

## Description

This PR adds a new configuration flag that shows/hides tabs in studio home along with some new functionality around to V1 and V2 Libraries.

When the new `LIBRARY_MODE` flag is set to `"mixed"` (default in dev) it will show "Libraries" and "Legacy Libraries" tabs that correspond to v1 and v2 tabs respectively:

<img width="1372" alt="Screenshot 2024-06-03 at 3 47 48 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/6829768/a7f39296-6fee-4a4b-8d54-338012834683">

<img width="1370" alt="Screenshot 2024-06-03 at 3 48 35 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/6829768/a39faac8-f4b9-44b7-9064-9d8f89a92ea6">

When the new `LIBRARY_MODE` flag is set to `"v1 only"` (default in production) or `"v2 only"`, only one tab "Libraries" is shown and only the respective libraries are fetched when the tab is clicked.

In addition to the above changes, the URL/route now updates when clicking on the tabs, and navigating to it directly would open up that tab as well as a new placeholder page that you will be redirected to when clicking on a v2 library if the library authoring MFE is not enabled:

<img width="1372" alt="Screenshot 2024-06-03 at 3 51 16 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/6829768/bfa6339f-8b47-48c3-8264-5a174989798c">

## Supporting information

Related Tickets:
- Depends on: https://github.com/openedx/edx-platform/pull/34879
- Related Issue: https://github.com/openedx/frontend-app-course-authoring/issues/1033

## Testing instructions

1. Run this PR's branch in your local env
1. Make sure you have the https://github.com/openedx/edx-platform/pull/34879/ running locally
1. Make sure you have the `studio.library_authoring_mfe` waffle flag enabled in the 
1. The "Libraries" and "Legacy Libraries" tabs appear in the studio home page
1. When clicking on the "Libraries" shows:
    1. v2 libraries are shown
    1. They should be paginated (you should see "Show x out of y", to further confirm pagination is working, set the default `pageSize` in https://github.com/openedx/frontend-app-course-authoring/pull/1050/files#diff-8eef2c5f738aa3151262ca0ca6634eb16c533410f9286ce96c0a58061590bac4R57 to a lower number, and confirm going through multiple pages of results)
    1. The route in the URL changes to `/libraries`
    1. When refreshing the page, it should keep you on the "Libraries" tab 
    1. When clicking on a v2 library it takes you to the library authoring mfe for that v2 library
    1. When clicking on "New Library" button it should take you to the library authoring mfe to create a new v2 library
    1. When disabling the `studio.library_authoring_mfe` waffle flag, and then clicking on a v2 library it should take you to the placeholder page (shown in screenshot above), the url should reflect the clicked v2 library (eg: `/library/lib:SampleTaxonomyOrg1:AL1`)
    1. The same should happen when clicking on "New Library" when `studio.library_authoring_mfe` waffle flag is disabled, it should redirect you to the placeholder page
1. When clicking on the "Legacy Libraries" tab:
    1. v1 libraries are shown
    1. The route in the URL changes to `/libraries-v1`
    1. When refreshing the page it should keep you on the "Legacy Libraries" tab 
    1. When clicking on a v1 library it takes you to the legacy studio UI for the clicked library
    1. Since the default `LIBRARY_MODE` is "mixed", clicking on "New Library" should still take you to the library authoring mfe to create a new v2 library
1. Change the `LIBRARY_MODE` [flag](https://github.com/openedx/frontend-app-course-authoring/pull/1050/files#diff-da4c41d59c967338247b2f0ea6d845ab05171f942fd9521b96d3304202185db9R49) to "v2 only" and restart the course-authoring container
    1. Confirm that there is only 1 tab "Library" that shows up
    1. When clicked it shows v2 libraries
    1. The route changes to `/libraries`  
    1. When clicking on "New Library" it takes you to the library authoring mfe if enabled to create new v2 libraries otherwise it takes you to a placeholder page
1. Change the `LIBRARY_MODE` [flag](https://github.com/openedx/frontend-app-course-authoring/pull/1050/files#diff-da4c41d59c967338247b2f0ea6d845ab05171f942fd9521b96d3304202185db9R49) to "v1 only" and restart the course-authoring container
    1. Confirm that there is only 1 tab "Library" that shows up
    1. When clicked it shows v1 libraries
    1.  The route changes to `/libraries-v1`
    1. When clicking "New Library" it takes you to the legacy UI to create a new v1 library
---
Private-ref: [FAL-3751](https://tasks.opencraft.com/browse/FAL-3751)
